index.js

```diff
@@ -1,48 +1,35 @@
-/* eslint-disable new-cap */
 'use strict';
-var dns = require('native-dns');
-var OPEN_DNS_IPV4 = '208.67.222.222';
-var OPEN_DNS_IPV6 = '2620:0:ccc::2';
+var dns = require('dns-socket');
 
 var type = {
 	v4: {
-		server: {
-			address: OPEN_DNS_IPV4,
-			port: 53,
-			type: 'udp'
-		},
-		question: dns.Question({
+		server: '208.67.222.222',
+		question: {
 			name: 'myip.opendns.com',
 			type: 'A'
-		})
+		}
 	},
 	v6: {
-		server: {
-			address: OPEN_DNS_IPV6,
-			port: 53,
-			type: 'udp'
-		},
-		question: dns.Question({
+		server: '2620:0:ccc::2',
+		question: {
 			name: 'myip.opendns.com',
 			type: 'AAAA'
-		})
+		}
 	}
 };
 
 function query(version, cb) {
-	var req = dns.Request(type[version]);
+	var data = type[version];
 
-	req.on('timeout', function () {
-		cb(new Error('Request timed out'));
-	});
-
-	req.on('message', function (err, res) {
+	dns().query({
```

Comment:
For v6 to work, we need to pass a v6 socket, dns-socket otherwise [uses a v4 socket](https://github.com/mafintosh/dns-socket/blob/4a5226637b3ddba91ab26c9389f68f7072cfff17/index.js#L17), like:

```js
dns({socket: dgram.createSocket(version === "v6" ? "udp6" : "udp4")})
```